plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2236,22 +2246,28 @@ public TableStatisticsMetadata getStatisticsCollectionMetadataForWrite(Connector
 
         Optional<Set<String>> analyzeColumnNames = Optional.empty();
         String tableLocation = getLocation(tableMetadata.getProperties());
+        Optional<ExtendedStatistics> existingStatistics = Optional.empty();
         if (tableLocation != null) {
-            analyzeColumnNames = statisticsAccess.readExtendedStatistics(session, tableLocation)
-                    .flatMap(ExtendedStatistics::getAnalyzedColumns);
+            existingStatistics = statisticsAccess.readExtendedStatistics(session, tableLocation);
+            analyzeColumnNames = existingStatistics.flatMap(ExtendedStatistics::getAnalyzedColumns);
         }
 
         return getStatisticsCollectionMetadata(
                 tableMetadata.getColumns(),
                 analyzeColumnNames.orElse(allColumnNames),
+                existingStatistics.isPresent(),
                 true);
     }
 
     private TableStatisticsMetadata getStatisticsCollectionMetadata(
             List<ColumnMetadata> tableColumns,
             Set<String> analyzeColumnNames,
+            boolean areExtendedStatisticsPresent,
             boolean isCollectionOnWrite)
     {
+        // Collect file statistics only when performing ANALYZE on a table without extended statistics
+        boolean collectFileStatistics = !areExtendedStatisticsPresent && !isCollectionOnWrite;
```

Comment:
Why?
Someone could have extended statistics (created by Trino 413) and want to ANALYZE table to collect also file-level stats.

Let's discuss and improve explanation in the code.

Reply:
The idea was to collect file-level statistics only for initial `ANALYZE` in this commit. Checking if extended statistics are empty is currently used to determine if it's initial statistics collection.

Reply:
For mentioned case maybe `drop_extended_stats` before ANALYZE (or force_recalculate_statistics with https://github.com/trinodb/trino/pull/16634) would be a easiest solution?